classify.c

```diff
@@ -604,19 +607,42 @@ static int check_for_irq_ban(int irq, GList *proc_interrupts)
 	return 0;
 }
 
+static void get_mod_name(char *path, char *mod)
+{
+	char *cmd;
+	FILE *output;
+
+	if (!path)
+		return;
+
+	cmd = alloca(strlen(path) + 30);
+	if (!cmd)
+		return;
+
+	sprintf(cmd, "cat %s/uevent | grep \"DRIVER=\"", path);
+	output = popen(cmd, "r");
+	if (!output)
+		return;
+
+	fgets(mod, 128, output);
+	pclose(output);
```

Comment:
why go to all the trouble of fishing the DRIVER field out of the event file?  you can find the same information via basename(readlink(/sys/bus/pci/*/driver)

Reply:
done